docs/articles/documentation/guides/advanced-guides/test-https-features-and-http2-websites.md

```diff
@@ -23,21 +25,95 @@ If the tested page does not use HTTPS-specific features (like
+    ```
+
+In **PowerShell**, [install OpenSSL](https://adamtheautomator.com/openssl-powershell/#Installing_OpenSSL_using_Chocolatey) and [update environment variables](https://adamtheautomator.com/openssl-powershell/#Update_PowerShell_Profile_Environment_Variables). Afterwards, proceed with the instructions [above](#generate-a-certificate-from-the-command-line).
+
+##### Run Tests
 
 When you run tests from the command line, use the [--ssl](../../reference/command-line-interface.md#--ssl-options) flag to enable HTTPS on a proxy server. Specify options required to initialize [a Node.js HTTPS server](https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener) after this flag in a semicolon-separated string. The most commonly used SSL options are described in the [TLS topic](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options) in the Node.js documentation.
 
-The example below uses the PFX encoded private key and certificate chain to create an HTTPS server.
+The examples below use a previously generated `.pfx` file to establish an HTTPS server.
+
+In ***Chrome*** and ***Chromium-Based*** Browsers use `--allow-insecure-localhost` flag and [`--hostname`](../../reference/command-line-interface.md#--hostname-name) CLI parameter to make Chrome trust the certificate.
+
+```sh
+testcafe chrome --allow-insecure-localhost --hostname localhost test.js --ssl pfx=path/to/certificate.pfx;rejectUnauthorized=true;
+```
+
+In ***Firefox*** it is necessary to add your self-signed root authority to the trusted list. Follow the steps:
+
+1. [Create a new profile](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles#w_creating-a-profile) in Firefox settings;
+2. [Run Firefox with this profile](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles#w_manage-profiles-when-firefox-is-open) and [import the CA certificate](https://support.mozilla.org/en-US/questions/1059377) (`myCA.pem`) you generated previously. Make sure to check the **"This certificate can identify websites"** checkbox;
+3. Load this new profile when TestCafe Studio launches Firefox. To do this, add `-P <profile_name>` to the browser's command line arguments. Ensure that **"Use the selected  user profile at startup without asking"** is disabled.
 
 ```sh
-testcafe --ssl pfx=path/to/file.pfx;rejectUnauthorized=true;...
+testcafe 'firefox -P testing-profile' --hostname localhost assertTest.js --ssl pfx=certificate.pfx;rejectUnauthorized=true;
 ```
 
-When you use the programming interface, pass the HTTPS server options to the [createTestCafe](../../reference/testcafe-api/global/createtestcafe.md) method.
+> The certificate created in the example is valid only for localhost domain, hence the `--hostname` option. If you wish to use another domain for testing, enter it in the `[alt_names]` section of `testdomain.ext` configuration file before creating the certificate.  
```

Comment:
'hence' is not a good fit for tech docs. `This is why the --hostname option is used` is better.